### PR TITLE
feat: add get drops entitlements endpoint to helix

### PR DESCRIPTION
**Pull request by iProdigy**
### Prerequisites for Code Changes
* [x] This pull request follows the code style of the project
* [ ] I have tested this feature

### Changes Proposed
* Implement `TwitchHelix#getDropsEntitlements(String, String, String, String, String, Integer)`

### Additional Information 
https://dev.twitch.tv/docs/api/reference#get-drops-entitlements (was added under a month ago to the API)

### CODE REVIEW

File: rest-helix/src/main/java/com/github/twitch4j/helix/domain/DropsEntitlementList.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+public class DropsEntitlementList {
+
+    /**
+     * The entitlements
+     */
+    private List<DropsEntitlement> data;
```
Comment (PhilippHeuer) [inline rest-helix/src/main/java/com/github/twitch4j/helix/domain/DropsEntitlementList.java:14]: There is no getter to access this

**Review by PhilippHeuer — CHANGES_REQUESTED** [review]
Looks good, i like the workaround you did in `getPaginationCursor` to make sure we can handle it in both cases :)

File: rest-helix/src/main/java/com/github/twitch4j/helix/domain/DropsEntitlementList.java (hunk 1 above)
Comment (iProdigy) [reply rest-helix/src/main/java/com/github/twitch4j/helix/domain/DropsEntitlementList.java:23]: Good catch! Added the proper annotations to the class

File: rest-helix/src/main/java/com/github/twitch4j/helix/domain/DropsEntitlementList.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+    /**
+     * If provided, is the key used to fetch the next page of data.
+     * If not provided, the current response is the last page of data available.
+     */
+    private Object pagination;
```
Comment (PhilippHeuer) [inline rest-helix/src/main/java/com/github/twitch4j/helix/domain/DropsEntitlementList.java:29]: Lets add `@Getter(AccessLevel.PRIVATE)` here as we have the method below to access the pagination curuser.

**Review by PhilippHeuer — APPROVED** [review]

**Pull request merged**